terraform/api_gateway.tf.json.template.py

```diff
@@ -370,18 +381,13 @@ def for_domain(cls, domain):
                 }
             },
             'aws_lambda_function_event_invoke_config': {
-                function_name: {
-                    'function_name': '${aws_lambda_function.%s.function_name}' % function_name,
-                    'maximum_retry_attempts': retry_attempts
+                f'{retry.lambda_name}_{retry.handler_name}': {
+                    'function_name': '${aws_lambda_function.%s.function_name}'
+                                     % f'{retry.lambda_name}_{retry.handler_name}',
+                    'maximum_retry_attempts': retry.value
                 }
-                for function_name, retry_attempts in
-                [
-                    (f'indexer_{lm}', 0)
-                    for lm in ['forward_alb_logs', 'forward_s3_logs']
-                    if config.enable_log_forwarding
-                ] + [
-                    (f'{lm}_{lm}cachehealth', 1) for lm in ['indexer', 'service']
-                ]
+                for app in apps
+                for retry in app.chalice.retries()
```

Comment:
@hannes-ucsc, this has the side-effect of creating a `aws_lambda_function_event_invoke_config` Terraform resource for all the other `EventSourceHandler` type of handlers, i.e. indexer_contribute, service_manifest, etc.
Is this desired?

If so, we may omit adding the `@retry` decorator to the forwarder Lambdas since it would be handled by the default retry value case.

Reply:
Something is wrong with the implementation of `app.chalice.retries`. The list it returns should only contain items for functions that use a retry count that's different from the default, which I believe is 0.

Reply:
I've opted for …
```
  value = getattr(handler, 'retries', None)
  if value is not None:
```
… as a [solution](https://github.com/DataBiosphere/azul/pull/6118/files#diff-a853ec0402888ff4ae4b573a31f9027350e2240fed755661acff11139e6c5c4bR556-R557) to this undesired behavior.